test/e2e/kubernetes/pod/pod.go

```diff
@@ -158,6 +159,70 @@ func RunLinuxPod(image, name, namespace, command string, printOutput bool) (*Pod
 	return p, nil
 }
 
+// RunWindowsPod will create a pod that runs a powershell command
+// --overrides := `"spec": {"nodeSelector":{"beta.kubernetes.io/os":"windows"}}}`
+func RunWindowsPod(image, name, namespace, command string, printOutput bool) (*Pod, error) {
+	overrides := `{ "spec": {"nodeSelector":{"beta.kubernetes.io/os":"windows"}}}`
+	cmd := exec.Command("kubectl", "run", name, "-n", namespace, "--image", image, "--image-pull-policy=IfNotPresent", "--restart=Never", "--overrides", overrides, "--command", "--", "powershell", command)
+	var out []byte
+	var err error
+	if printOutput {
+		out, err = util.RunAndLogCommand(cmd)
+	} else {
+		out, err = cmd.CombinedOutput()
+	}
+	if err != nil {
+		log.Printf("Error trying to deploy %s [%s] in namespace %s:%s\n", name, image, namespace, string(out))
+		return nil, err
+	}
+	p, err := Get(name, namespace)
+	if err != nil {
+		log.Printf("Error while trying to fetch Pod %s in namespace %s:%s\n", name, namespace, err)
+		return nil, err
+	}
+	return p, nil
+}
+
+type podRunnerCmd func(string, string, string, string, bool) (*Pod, error)
+
+// RunCommandMultipleTimes runs the same command 'attempts' times
+func RunCommandMultipleTimes(podRunnerCmd podRunnerCmd, image, name, command string, attempts int) (int, error) {
+	var successes int
+	for i := 0; i < attempts; i++ {
+		r := rand.New(rand.NewSource(time.Now().UnixNano()))
+		podName := fmt.Sprintf("%s-%d", name, r.Intn(99999))
+		var p *Pod
+		var err error
+		if i < 1 {
+			// Print the first attempt
+			p, err = podRunnerCmd(image, podName, "default", command, true)
+		} else {
+			p, err = podRunnerCmd(image, podName, "default", command, false)
+		}
+		if err != nil {
+			return successes, err
+		}
+		succeeded, _ := p.WaitOnSucceeded(1*time.Second, 2*time.Minute)
+		cmd := exec.Command("kubectl", "logs", podName, "-n", "default")
+		out, err := util.RunAndLogCommand(cmd)
+		if err != nil {
+			log.Printf("Unable to get logs from pod %s\n", podName)
+		} else {
+			log.Printf("%s\n", string(out[:]))
+		}
+		if succeeded {
+			successes++
+		}
+		err = p.Delete()
+		if err != nil {
+			return successes, err
+		}
+	}
+
+	log.Printf("Ran command successfully on %d of %d test attempts\n\n", successes, attempts)
```

Comment:
I think this would be good to `defer` logging the summary to consistently log how many iterations were attempted, desired, and succeeded even on failure. It would be good to know if there's a pattern that leads to failing at N of M attempts. On a failed run it won't say how many attempts were made, just that one of them failed

Reply:
Great idea.  I have updated it to defer and log the successful attempts, actual attempts, and desired attempts